### DISCUSSION TITLE
[Need help] How to use snap7.util.get_int?

### DISCUSSION THREAD

**Discussion by tom9672** [gijzelaerr/python-snap7#448] [Q&A] [answered]
I'm new to snap7. I read the document [https://python-snap7.readthedocs.io/en/latest/API/client.html](url), and find use `db_read(db_number: int, start: int, size: int) → bytearray` to read data from PLC, as the example ` client.db_read(1, 10, 4) #bytearray(b'\x00\x00') `  shows read db1 from 10 to 14, why the size result is 2? But another example at the top of this document, the size of read result is `client.db_read(1, 0, 4) #bytearray(b"\x00\x00\x00\x00")` and the read size is the same (4). 

What needs to do if I would like to read a db_area as int type, I find `snap7.util.get_int(bytearray_: bytearray, byte_index: int) → int` seems to convert bytearray to int.

I tried:
```python
import snap7
b = bytearray(b"\x01\x18")
i = snap7.util.get_int(b,0)  #280

b = bytearray(b"\x01\x18\x01")
i = snap7.util.get_int(b,0)  #280

b = bytearray(b"\x01\x18\x01")
i = snap7.util.get_int(b,1)  #6145
```

That seems mean `get_int` takes 2 bytes as one int. Can I consider if I need to read 20 byte size if I need 10 int, and to convert these 20 bytes, I need to loop the bytearray and use 10 times `get_int`?

Is there a better way to read int from a db area.

Thanks in advance!

**Comment by tom9672** [comment]
I fond the function `snap7.util.get_array(bytearray_: bytearray, byte_index: int)->List`, seems not implemented yet.

Is this function aims to convert bytearray to a list result? Or theres another function to do so.


Here's a function to convert the whole bytearray_ to int list.

```python
def get_array_int(bytearray_: bytearray) -> List[int]:
    if len(bytearray_) % 2 == 1: return []
    
    result = []
    for i in range(0, len(bytearray_), 2):
        integer = (bytearray_[i] << 8) + bytearray_[i+1]
        result.append(integer)
    return result
```

**Comment by nikteliy** [comment]
Hi, @tom9672 
> as the example client.db_read(1, 10, 4) #bytearray(b'\x00\x00') shows read db1 from 10 to 14, why the size result is 2?

It's a typo. Thanks for pointing this out.

> What needs to do if I would like to read a db_area as int type

If you want to convert a bytearray to integers, you can use the built-in [struct](https://docs.python.org/3/library/struct.html) module.  Here is an example:
```python
>>> import struct
>>> bytearray_ = b'\x00\x01\x00\x16\x01M\x11\\09' # (1, 22, 333, 4444, 12345)
>>> tuple_of_ints = struct.unpack('>5H', bytearray_)
>>> print(tuple_of_ints)
(1, 22, 333, 4444, 12345)
>>> a, b, c, d, e = struct.unpack('>5H', bytearray_) # unpack to variables
>>> print(a, b, c, d, e)
1 22 333 4444 12345
```

**Comment by gijzelaerr** [comment, marked as answer]
Closing as answered. The documentation typo was acknowledged, and the use of `struct.unpack` for batch integer conversion was explained.
